# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsDOMWindowUtils.cpp
# Commit: 415c7e1ea941
# Full Hash: 415c7e1ea941ca3bc0b33c5d91fe9782044ee9a0
# Author: Timothy Nikkel <tnikkel@gmail.com>
# Date: 2022-02-12 21:25:59
# Regressor Bug: 1754436
# File Overlap Count: 2
# Description:
#   Bug 1754662. Refactor nsLayoutUtils::GetPopupFrameForEventCoordinates. r=emilio
#   
#   This lets us re-use code for bug 1754436.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138392
# ==============================================================================

diff -r 152c2969ca06 -r 415c7e1ea941 dom/base/nsDOMWindowUtils.cpp
--- a/dom/base/nsDOMWindowUtils.cpp	Sat Feb 12 04:34:08 2022 -0500
+++ b/dom/base/nsDOMWindowUtils.cpp	Sat Feb 12 09:41:50 2022 +0000
@@ -2378,11 +2378,8 @@
 
   if (message == eQueryCharacterAtPoint) {
     // Looking for the widget at the point.
-    WidgetQueryContentEvent dummyEvent(true, eQueryContentState, widget);
-    dummyEvent.Init(options);
-    InitEvent(dummyEvent, &pt);
-    nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForEventCoordinates(
-        presContext->GetRootPresContext(), &dummyEvent);
+    nsIFrame* popupFrame = nsLayoutUtils::GetPopupFrameForPoint(
+        presContext->GetRootPresContext(), widget, pt);
 
     LayoutDeviceIntRect widgetBounds = widget->GetClientBounds();
     widgetBounds.MoveTo(0, 0);